columnq/src/table/database.rs

```diff
@@ -0,0 +1,74 @@
+use connectorx::prelude::*;
+use connectorx::sources::mysql::BinaryProtocol;
+use datafusion::arrow::record_batch::RecordBatch;
+use log::debug;
+
+use crate::table::TableSource;
+
+pub enum DatabaseLoader {
+    MySQL,
+    SQLite,
+}
+
+impl DatabaseLoader {
+    pub async fn to_mem_table(
+        &self,
+        t: &TableSource,
+    ) -> anyhow::Result<datafusion::datasource::MemTable> {
```

Comment:
anyhow is supposed to be used in end application code, not in libraries because it will not be easy for library user to match and react to a specific error, see https://github.com/dtolnay/anyhow#comparison-to-thiserror